File: kata-6kyu/easy-balance-checking.js
function balance(book) {
  book = book.replace(/[!=:;{}?,]/g,'')
             .split('\n')
             .filter(el=>el)
             .map((el,i)=>!i ? el : el.split(' ')); 
  const body = ['Original Balance: ' + (+book[0]).toFixed(2)];
  const originalBalance = book[0];
  let balance = book[0];
  for(let i = 1; i < book.length; i++){
    balance = (balance - book[i][2]).toFixed(2);
    body.push(book[i].map((el,i)=> i<2 ? el : (+el).toFixed(2)).join(' ') + ' Balance '+ balance);
  }
  const expense = (originalBalance - balance).toFixed(2);
  body.push('Total expense  ' + expense);
  body.push('Average expense  ' + (expense / (book.length-1)).toFixed(2));
  return body.join('\r\n')
}

File: python-kata-6kyu/easy-balance-checking.py
def money_form(n):
    n = format(n, '.2f') + '0'
    return n[:n.index('.')+3]

def balance(book):
    new_book = ''
    for c in book:
        new_book+= c * (c not in '!=,:;{}?')
        
    arr = new_book.split('\n')
    
    original_balance = float(arr[0])
    balance = original_balance
    arr[0] = f"Original Balance: {money_form(original_balance)}\r"
    
    while '' in arr:    
        arr.remove('')

    for i,s in enumerate(arr[1:]):
        sub_arr = s.split(' ')
        price = float(sub_arr[2])
        balance-= price
        arr[i+1]= f"{' '.join(sub_arr[:-1])} {money_form(price)} Balance {money_form(balance)}\r"
        expense = original_balance - balance
    return "\n".join(arr) + f"\nTotal expense  {money_form(expense)}\r\nAverage expense  {money_form(expense/(len(arr)-1))}"
